frontend/app/controllers/TierController.scala

```diff
@@ -11,10 +11,11 @@ import play.api.libs.concurrent.Execution.Implicits.defaultContext
 import play.api.libs.json.Json
 import play.api.mvc.{AnyContent, Result, Controller, DiscardingCookie}
 import services.{SubscriptionService, IdentityApi, IdentityService, MemberService}
+import tracking.ActivityTracking
 
 import scala.concurrent.Future
 
-trait DowngradeTier {
+trait DowngradeTier extends ActivityTracking {
```

Comment:
Me again. I think the `extractCampaignCode` should be defined in this file. It's the only place that uses it? 
Unless you plan to use it somewhere else? In that case I recommend a separate file to `ActivityTracking`.